### PR TITLE
KeyboardAvoidingView: return KeyboardFrame height directly when botto…

### PR DESCRIPTION
**Pull request by EllisShen**
KeyboardAvoidingView didn't handle a specific situation. On Android, when a user selects the input method from ENG to Google Voice, the text input area would be covered by Google Voice view.

Image of Issue

Basically, `relativeKeyboardHeight` is assuming that `onKeyboardChange` event is triggered from normal state to the virtual keyboard state. However, there's a situation that user changes the input method to Google Voice over from English input. It would be miscalculated by `relativeKeyboardHeight` function and results in a wrong padding Bottom.

My suggestion is that adding a statement in `relativeKeyboardHeight` to determine if the View is already padded at the Bottom. If it's, return the keyboard.height directly.

### CODE REVIEW

**Comment by EllisShen** [comment]
@nicklockwood and @davidaurelio, could you review my commit?

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js [hunk 1]
```diff
@@ -88,10 +88,15 @@ const KeyboardAvoidingView = React.createClass({
 
   relativeKeyboardHeight(keyboardFrame: ScreenRect): number {
     const frame = this.frame;
+    const {bottom} = this.state;
```
Comment (brentvatne) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: nit: `const { bottom } = this.state;`

Comment (brentvatne) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: actually it seems like this file uses the format that you used here :\

**Comment by brentvatne** [comment]
@EllisShen - have you tested this on iOS to ensure it doesn't break anything?

File: Libraries/Components/Keyboard/KeyboardAvoidingView.js (hunk 1 above)
Comment (EllisShen) [inline Libraries/Components/Keyboard/KeyboardAvoidingView.js]: @brentvatne, yes, it is... do you want me to revise the whole file to follow the standard?

**Comment by EllisShen** [comment]
@brentvatne, actually I haven't test it on iOS. I will let you know my verification result asap.

**Comment by hramos** [comment]
Ping us when you've tested on iOS.

**Comment by mkonicek** [comment]
Adding needs-revision because of missing Test Plan. See: 
https://github.com/facebook/react-native/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by lacker** [comment]
Hey, I think this pull request has been abandoned so I am going to close it. If you are still interested in working on this please feel free to reopen it!

**Pull request closed without merging**